src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,130 @@
+}
+
+class PlannerPlanAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.PLANNER_PLAN_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'createdDateTime', 'owner'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.ownerGroupId) {
+      this.createPlan(args.options.ownerGroupId, args, logger, cb);
+    }
+    else {
+      this.getGroupId(args)
+        .then((groupId: string): void => {
+          this.createPlan(groupId, args, logger, cb);
+        }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+    }
+  }
+
+  private createPlan(groupId: string, args: CommandArgs, logger: Logger, cb: () => void) {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      },
+      responseType: 'json',
+      data: {
+        "owner": groupId,
+        "title": args.options.title
+      }
+    };
+    request
+      .post(requestOptions)
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: any[] }>(requestOptions)
+      .then(response => {
+        const group: any | undefined = response.value[0];
+
+        if (!group) {
+          return Promise.reject(`The specified owner group does not exist`);
+        }
+
+        return Promise.resolve(group.id);
+      });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-t, --title <title>'
+      },
+      {
+        option: "--ownerGroupId [ownerGroupId]"
+      },
+      {
+        option: "--ownerGroupName [ownerGroupName]"
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (!args.options.ownerGroupId && !args.options.ownerGroupName) {
+      return 'Specify either ownerGroupId or ownerGroupName but not both';
```

Comment:
This error message is incorrect. In this case, the user hasn't specified either one option